Objects/longobject.c

```diff
@@ -6773,3 +6774,114 @@ int PyLong_AsUInt64(PyObject *obj, uint64_t *value)
 {
     LONG_TO_UINT(obj, value, "C uint64_t");
 }
+
+
+static const PyLongLayout PyLong_LAYOUT = {
+    .bits_per_digit = PyLong_SHIFT,
+    .digits_order = -1,  // least significant first
+    .endian = PY_LITTLE_ENDIAN ? -1 : 1,
+    .digit_size = sizeof(digit),
+};
+
+
+const PyLongLayout* PyLong_GetNativeLayout(void)
+{
+    return &PyLong_LAYOUT;
+}
+
+
+int
+PyLong_Export(PyObject *obj, PyLongExport *export_long)
+{
+    if (!PyLong_Check(obj)) {
+        PyErr_Format(PyExc_TypeError, "expect int, got %T", obj);
+        return -1;
+    }
+
+    int overflow;
+#if SIZEOF_LONG == 8
+    long value = PyLong_AsLongAndOverflow(obj, &overflow);
+#elif SIZEOF_LONG_LONG == 8
+    long long value = PyLong_AsLongLongAndOverflow(obj, &overflow);
+#else
+#   error "unable to convert a long to int64_t"
```

Comment:
Hmm, I'm not sure that this is reachable on some system, but why fail?!

long long should fit int64_t, so you can just keep L6805.

Reply:
> Hmm, I'm not sure that this is reachable on some system, but why fail?!

If long and long long are larger than 8 bytes on an hypothetical platform, I prefer to fail than producing invalid code.

Reply:
Right:(  Then maybe change ``==`` to ``<=`` and try long long first?

Reply:
I'm not sure which problem you're trying to solve. My code should work on all platforms.

Reply:
> My code should work on all platforms.

Hmm, I've tried to add this as a backport and got failures: https://github.com/aleaxit/gmpy/actions/runs/10922014588/job/30315384183.  Previous (successful) commit was with just PyLong_AsLongAndOverflow().

Reply:
I modified the code to replace `#error` with a `Py_BUILD_ASSERT()`.